src/utility.ts

```diff
@@ -12,62 +17,65 @@ export const Utility = {
 	uint8ArrayToString(input: Uint8Array): string {
 		return new TextDecoder().decode(input);
 	},
-	encodeBase64(input: string): string {
-		return Buffer.from(input).toString("base64");
-	},
-
-	decodeBase64(b64String: string): string {
-		return Buffer.from(b64String).toString("binary");
-	},
-
-	filterString<T>(filter: Filter | LogicalFilter): string {
+	filterToString<T>(filter: SelectorFilter<T> | LogicalFilter<T>): string {
 		// eslint-disable-next-line no-prototype-builtins
 		return filter.hasOwnProperty("logicalOperator")
-			? Utility._logicalFilterString(filter as LogicalFilter)
-			: this.objToJsonString(this.filterJSON(filter as Filter));
+			? Utility._logicalFilterToString(filter as LogicalFilter<T>)
+			: this.objToJsonString(this._selectorFilterToJSONObj(filter as SelectorFilter<T>));
 	},
-
-	filterJSON(filter: Filter): Filters {
-		return {  [filter.key]: filter.val  };
+	_selectorFilterToString<T extends TigrisCollectionType>(filter: SelectorFilter<T>): string {
+		if (filter.op == SelectorFilterOperator.EQ) {
+			return Utility.objToJsonString(Utility._flattenObj(Utility._selectorFilterToJSONObj(filter)))
+		}
+		return "";
 	},
-
-	_logicalFilterString(filter: LogicalFilter): string {
-		return this.objToJsonString(this.logicalFilterJSON(filter));
+	_selectorFilterToJSONObj<T>(filter: SelectorFilter<T>): object {
+		if (filter.op == SelectorFilterOperator.EQ) {
+			return filter.fields
+		}
+		// add support later
+		return {}
 	},
-
-	logicalFilterJSON(filter: LogicalFilter): LogicalFilters {
-		const { filters, logicalFilters, logicalOperator } = filter;
-
-		const filtersArray = filters?.map(Utility.filterJSON) ?? [];
-		const logicalFiltersArray = logicalFilters?.map(Utility.logicalFilterJSON) ?? [];
-
-		return {
-			[logicalOperator]: [...filtersArray, ...logicalFiltersArray],
-		};
+	_logicalFilterToString<T>(filter: LogicalFilter<T>): string {
+		return this.objToJsonString(Utility._logicalFilterToJSONObj(filter))
+	},
+	_logicalFilterToJSONObj<T>(filter: LogicalFilter<T>): object {
+		const result = {};
+		const innerFilters = [];
+		result[filter.op] = innerFilters;
+		if (filter.selectorFilters) {
+			for (const value of filter.selectorFilters) {
+				// @ts-ignore
```

Comment:
question: why you had to add ts-ignore?

Reply:
Good catch. that was added in intermediate development to get around some errors and it was left to clean up. Removed it.